examples/cpp/dds/CustomPayloadPoolExample/CustomPayloadPoolDataSubscriber.cpp

```diff
@@ -0,0 +1,164 @@
+
+    if (participant_ == nullptr)
+    {
+        return false;
+    }
+
+    //REGISTER THE TYPE
+    type_.register_type(participant_);
+
+    //CREATE THE SUBSCRIBER
+    SubscriberQos sqos = SUBSCRIBER_QOS_DEFAULT;
+
+    subscriber_ = participant_->create_subscriber(sqos, nullptr);
+
+    if (subscriber_ == nullptr)
+    {
+        return false;
+    }
+
+    //CREATE THE TOPIC
+    TopicQos tqos = TOPIC_QOS_DEFAULT;
+
+    topic_ = participant_->create_topic(
+        "CustomPayloadPoolTopic",
+        "CustomPayloadPoolData",
+        tqos);
+
+    if (topic_ == nullptr)
+    {
+        return false;
+    }
+
+    // CREATE THE READER
+    DataReaderQos rqos = DATAREADER_QOS_DEFAULT;
+    rqos.reliability().kind = RELIABLE_RELIABILITY_QOS;
+
+    reader_ = subscriber_->create_datareader(topic_, rqos, this, StatusMask::all(), payload_pool_);
+
+    if (reader_ == nullptr)
+    {
+        return false;
+    }
+
+    return true;
+}
+
+CustomPayloadPoolDataSubscriber::~CustomPayloadPoolDataSubscriber()
+{
+    if (reader_ != nullptr)
+    {
+        subscriber_->delete_datareader(reader_);
+    }
+    if (topic_ != nullptr)
+    {
+        participant_->delete_topic(topic_);
+    }
+    if (subscriber_ != nullptr)
+    {
+        participant_->delete_subscriber(subscriber_);
+    }
+    DomainParticipantFactory::get_instance()->delete_participant(participant_);
+}
+
+void CustomPayloadPoolDataSubscriber::on_subscription_matched(
+        DataReader*,
+        const SubscriptionMatchedStatus& info)
+{
+    if (info.current_count_change == 1)
+    {
+        matched_ = info.total_count;
+        std::cout << "Subscriber matched." << std::endl;
+    }
+    else if (info.current_count_change == -1)
+    {
+        matched_ = info.total_count;
+        std::cout << "Subscriber unmatched." << std::endl;
+    }
+    else
+    {
+        std::cout << info.current_count_change
+                  << " is not a valid value for SubscriptionMatchedStatus current count change" << std::endl;
+    }
+}
+
+void CustomPayloadPoolDataSubscriber::on_data_available(
+        DataReader* reader)
+{
+    SampleInfo info;
+    if (reader->take_next_sample(&hello_, &info) == ReturnCode_t::RETCODE_OK)
+    {
+        if (info.instance_state == ALIVE_INSTANCE_STATE)
+        {
+            samples_++;
+            // Print your structure data here.
+            std::cout << "Message " << hello_.message() << " " << hello_.index() << " RECEIVED" << std::endl;
+        }
+    }
+}
+
+void CustomPayloadPoolDataSubscriber::run()
+{
+    std::cout << "Subscriber running. Please press enter to stop the Subscriber" << std::endl;
+    std::cin.ignore();
+}
+
+void CustomPayloadPoolDataSubscriber::run(
+        uint32_t number)
+{
+    std::cout << "Subscriber running until " << number << "samples have been received" << std::endl;
+    while (number > samples_)
+    {
+        std::this_thread::sleep_for(std::chrono::milliseconds(500));
+    }
+}
```

Comment:
Mix these two into one and close with SIGINT. Furthermore, please use condition variables instead of sleeps